Add FullStory Analytics Components

cbauerman opened the pull request:
#### Changes proposed in this Pull Request

* Add new Component, `<FullStoryRecorder/>` to Calypso
   *  Triggers tacking for logged in, US users, when rendered ( and `record` property not set to false )
* Add new Component, `<AnalyticsSafeContainer/>` to protect sensitive user information
   * In addition to FullStory this should implement safety for any other new analytics packages used
   * Care should be user to make all areas that could need protecting are found and covered
 

#### Testing instructions
 1. Add `FullStoryRecorder` and `AnalyticsSafeContainer` to multiple pages of a flow ( see my example at #52648 )
 2. Run in development mode
 3. Verify recording the flow works in the FullStory dashboard

cbauerman committed:
move FullStory init behind US users

cbauerman committed:
move FullStory constants to constant folder

cbauerman committed:
Remove import leftover from rebase

cbauerman committed:
Add devMode and recording control

simison commented:
I remember @tyxla implemented some generic libraries to add these scripts when we started using [HotJar](https://github.com/Automattic/wp-calypso/search?q=hotjar). Perhaps this could go to that same library to avoid boilerplate?

tyxla commented:
> I remember @tyxla implemented some generic libraries to add these scripts when we started using [HotJar](https://github.com/Automattic/wp-calypso/search?q=hotjar). Perhaps this could go to that same library to avoid boilerplate?

In case that helps, @simison is talking about this one: https://github.com/Automattic/wp-calypso/pull/30809

razvanpapadopol commented:
Can we use FullStory also in wp-admin? This would involve running the script on `[site_slug].wordpress.com` sub-domains instead of on `wordpress.com`.
I'm asking in case we might want to use it for screens rendered inside the editor like Welcome tour, Page picker, Focused launch.
cc: @simison

In case the points above are true, it makes sense to add the script to `calypso-analytics` [package](https://github.com/Automattic/wp-calypso/tree/trunk/packages/calypso-analytics) and import it back in `client` and `apps/editing-toolkit`.

Without access to `client/state`, we can use the `User` store from [data-stores package](https://github.com/Automattic/wp-calypso/tree/trunk/packages/data-stores) to get the country code of current user. 
Typing `wp.data.select( 'automattic/user' ).getCurrentUser().user_ip_country_code` in the JS console when in https://wordpress.com/new should return the same data as `getCurrentUserCountryCode( state )` does in Calypso.

simison commented:
Yep, we'll need this in wp-admin for both simple and Atomic sites, as well sites with custom domains (so not just `*.wordpress.com` subdomains). "Half of .com" is now on wp-admin. cc @obenland

jaswrks commented on client/lib/analytics/ad-tracking/constants.js:
```diff
@@ -69,6 +69,7 @@ export const TRACKING_IDS = {
 	bingInit: '4074038',
 	facebookInit: '823166884443641',
 	facebookJetpackInit: '919484458159593',
+	fullStoryOrgId: '1149DX',
```
Should this be `120RG4` ?

cbauerman committed:
Add `fullstory` feature flag

cbauerman committed:
Add FullStory init, AnalyticsSafeContainer

# Conflicts:
#	client/boot/common.js
#	client/my-sites/checkout/composite-checkout/composite-checkout.tsx

cbauerman committed:
Add FullStoryRecorder component

# Conflicts:
#	client/boot/common.js

cbauerman committed:
Move FullStoryInit to recorder component

# Conflicts:
#	client/boot/common.js

cbauerman committed:
Refine FullStoryRecorder

# Conflicts:
#	client/my-sites/backup/rewind-flow/download.tsx

cbauerman committed:
import changes from trunk

cbauerman committed:
re-add fullstoy package

cbauerman committed:
Add checkout-system-decider.js from trunk

cbauerman committed:
Change to correct OrgId

cbauerman commented:
⚠️ Branch re-based from trunk, local versions will need to be updated ⚠️

cbauerman committed:
Add component READMEs

cbauerman commented:
Closed in favor of #52768

cbauerman commented:
closed in favor of #52768
